EvolutionModel/WateringHole.cs

```diff
@@ -25,17 +25,20 @@ public class WateringHole : IFoodContainer, IEatable {
 
     public bool CanBeEatenBy(Species eater) => !eater.IsPredator && FoodAmount > 0;
 
-    public int GetsEaten(Species eater) {
-      if (CanBeEatenBy(eater) == true) {
-        if (FoodAmount > 1) {
-          FoodAmount--;
-          return 1;
-        } else {
-          throw new InvalidOperationException($"The {nameof(WateringHole)} is empty!");
+        public bool GetsEaten(bool eater, out int foodEaten)
```

Comment:
De guidelines voor het TryDo pattern (https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/exceptions-and-performance?fbclid=IwAR1_EPCO7SOritX_Grteus3ulF47jfGLGXyerIkq6RsRSUJmcpiK06HqD3A) zijn als volgt:

- [ ] DO use the prefix "Try" and Boolean return type for methods implementing this pattern.

- [ ] DO provide an exception-throwing member for each member using the Try-Parse Pattern.

Dit doen we op het moment niet. Die eerste is een naam dingetje. Methods met dit patroon moeten met `Try` beginnen in hun naam, om duidelijk te maken dat dat de manier is dat het werkt. `TryGetsEaten` bekt voor geen meter. Dus misschien renamen naar `TryEat`. 

Het tweede punt houdt in dat er ook een method moet blijven bestaand die wel exceptions gooit. Dus we willen uiteindelijk beiden hebben. Een `Eat` en een `TryEat`. Deze moeten allebei ook in de interface terechtkomen.